llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -3907,7 +3915,9 @@ bool SIInstrInfo::hasUnwantedEffectsWhenEXECEmpty(const MachineInstr &MI) const
   // However, executing them with EXEC = 0 causes them to operate on undefined
```

Comment:
I notice this comment is only really accurate for V_READFIRSTLANE which depends on EXEC.
readlane and writelane are well defined with EXEC=0 -- just we have other reasons for not running them.
If we are sure we want to include spill/reload here then they should probably be documented with a separate comment, because really we are avoiding these in EXEC=0 because it means a code branch should be run for *any* scalar operation.

Reply:
You are right. The readlane & writelane instructions shouldn't be here.
Posted https://github.com/llvm/llvm-project/pull/70206.